x-pack/plugins/maps/public/react_embeddable/map_react_embeddable.tsx

```diff
@@ -0,0 +1,228 @@
+import { ReactEmbeddableFactory, VALUE_CLICK_TRIGGER } from '@kbn/embeddable-plugin/public';
+import { EmbeddableStateWithType } from '@kbn/embeddable-plugin/common';
+import {
+  apiIsOfType,
+  getUnchangingComparator,
+  initializeTimeRange,
+  initializeTitles,
+  useBatchedPublishingSubjects,
+} from '@kbn/presentation-publishing';
+import { BehaviorSubject } from 'rxjs';
+import { apiPublishesSettings } from '@kbn/presentation-containers/interfaces/publishes_settings';
+import { MAP_SAVED_OBJECT_TYPE } from '../../common/constants';
+import { inject } from '../../common/embeddable';
+import type { MapApi, MapSerializedState } from './types';
+import { SavedMap } from '../routes/map_page';
+import { initializeReduxSync } from './initialize_redux_sync';
+import {
+  getByReferenceState,
+  getByValueState,
+  initializeLibraryTransforms,
+} from './library_transforms';
+import { getEmbeddableEnhanced, getSpacesApi } from '../kibana_services';
+import { initializeActionHandlers } from './initialize_action_handlers';
+import { MapContainer } from '../connected_components/map_container';
+import { waitUntilTimeLayersLoad$ } from '../routes/map_page/map_app/wait_until_time_layers_load';
+import { initializeCrossPanelActions } from './initialize_cross_panel_actions';
+import { initializeDataViews } from './initialize_data_views';
+import { initializeFetch } from './initialize_fetch';
+import { initializeEditApi } from './initialize_edit_api';
+import { extractReferences } from '../../common/migrations/references';
+
+export function getControlledBy(id: string) {
+  return `mapEmbeddablePanel${id}`;
+}
+
+export const mapEmbeddableFactory: ReactEmbeddableFactory<MapSerializedState, MapApi> = {
+  type: MAP_SAVED_OBJECT_TYPE,
+  deserializeState: (state) => {
+    return state.rawState
+      ? (inject(
+          state.rawState as EmbeddableStateWithType,
+          state.references ?? []
+        ) as unknown as MapSerializedState)
+      : {};
+  },
+  buildEmbeddable: async (state, buildApi, uuid, parentApi) => {
+    const savedMap = new SavedMap({
+      mapSerializedState: state,
+    });
+    await savedMap.whenReady();
+
+    // eslint bug, eslint thinks api is never reassigned even though it is
+    // eslint-disable-next-line prefer-const
+    let api: MapApi | undefined;
+    const getApi = () => api;
+    const sharingSavedObjectProps = savedMap.getSharingSavedObjectProps();
+    const spaces = getSpacesApi();
+    const controlledBy = getControlledBy(uuid);
+    const title = initializeTitles(state);
+    const timeRange = initializeTimeRange(state);
+    const dynamicActionsApi = getEmbeddableEnhanced()?.initializeReactEmbeddableDynamicActions(
+      uuid,
+      () => title.titlesApi.panelTitle.getValue(),
+      state
+    );
+    const maybeStopDynamicActions = dynamicActionsApi?.startDynamicActions();
+
+    const defaultPanelTitle$ = new BehaviorSubject<string | undefined>(
+      savedMap.getAttributes().title
+    );
```

Comment:
nit

A map that is saved to the library from the Maps editor (though not from "Save to library" panel action) can have a description. Can we add a `defaultPanelDescription$` subject as well?